src/addr_families.c

```diff
@@ -0,0 +1,832 @@
+    if (strcaseeq(family_name, "QIPCRTR", (size_t)family_name_len))
+        return WOLFSENTRY_AF_QIPCRTR;
+    if (strcaseeq(family_name, "SMC", (size_t)family_name_len))
+        return WOLFSENTRY_AF_SMC;
+    if (strcaseeq(family_name, "XDP", (size_t)family_name_len))
+        return WOLFSENTRY_AF_XDP;
+    if (strcaseeq(family_name, "IMPLINK", (size_t)family_name_len))
+        return WOLFSENTRY_AF_IMPLINK;
+    if (strcaseeq(family_name, "PUP", (size_t)family_name_len))
+        return WOLFSENTRY_AF_PUP;
+    if (strcaseeq(family_name, "CHAOS", (size_t)family_name_len))
+        return WOLFSENTRY_AF_CHAOS;
+    if (strcaseeq(family_name, "NETBIOS", (size_t)family_name_len))
+        return WOLFSENTRY_AF_NETBIOS;
+    if (strcaseeq(family_name, "ISO", (size_t)family_name_len))
+        return WOLFSENTRY_AF_ISO;
+    if (strcaseeq(family_name, "OSI", (size_t)family_name_len))
+        return WOLFSENTRY_AF_OSI;
+    if (strcaseeq(family_name, "ECMA", (size_t)family_name_len))
+        return WOLFSENTRY_AF_ECMA;
+    if (strcaseeq(family_name, "DATAKIT", (size_t)family_name_len))
+        return WOLFSENTRY_AF_DATAKIT;
+    if (strcaseeq(family_name, "DLI", (size_t)family_name_len))
+        return WOLFSENTRY_AF_DLI;
+    if (strcaseeq(family_name, "LAT", (size_t)family_name_len))
+        return WOLFSENTRY_AF_LAT;
+    if (strcaseeq(family_name, "HYLINK", (size_t)family_name_len))
+        return WOLFSENTRY_AF_HYLINK;
+    if (strcaseeq(family_name, "LINK", (size_t)family_name_len))
+        return WOLFSENTRY_AF_LINK;
+    if (strcaseeq(family_name, "COIP", (size_t)family_name_len))
+        return WOLFSENTRY_AF_COIP;
+    if (strcaseeq(family_name, "CNT", (size_t)family_name_len))
+        return WOLFSENTRY_AF_CNT;
+    if (strcaseeq(family_name, "SIP", (size_t)family_name_len))
+        return WOLFSENTRY_AF_SIP;
+    if (strcaseeq(family_name, "SLOW", (size_t)family_name_len))
+        return WOLFSENTRY_AF_SLOW;
+    if (strcaseeq(family_name, "SCLUSTER", (size_t)family_name_len))
+        return WOLFSENTRY_AF_SCLUSTER;
+    if (strcaseeq(family_name, "ARP", (size_t)family_name_len))
+        return WOLFSENTRY_AF_ARP;
+    if (strcaseeq(family_name, "IEEE80211", (size_t)family_name_len))
+        return WOLFSENTRY_AF_IEEE80211;
+    if (strcaseeq(family_name, "INET_SDP", (size_t)family_name_len))
+        return WOLFSENTRY_AF_INET_SDP;
+    if (strcaseeq(family_name, "INET6_SDP", (size_t)family_name_len))
+        return WOLFSENTRY_AF_INET6_SDP;
+    if (strcaseeq(family_name, "HYPERV", (size_t)family_name_len))
+        return WOLFSENTRY_AF_HYPERV;
+
+    if (errcode != NULL)
+        *errcode = WOLFSENTRY_ERROR_ENCODE(ITEM_NOT_FOUND);
+    return WOLFSENTRY_AF_UNSPEC;
+}
+
+static const char *wolfsentry_addr_family_ntop_1(
+    wolfsentry_addr_family_t family,
+    wolfsentry_errcode_t *errcode)
+{
+    if (errcode != NULL)
+        *errcode = WOLFSENTRY_ERROR_ENCODE(OK);
+    switch(family) {
+    case WOLFSENTRY_AF_UNSPEC:
```

Comment:
Likewise I think this might work as a LUT?

Reply:
likewise agree.